llvm/include/llvm/Analysis/DXILResource.h

```diff
@@ -212,6 +216,53 @@ class ResourceInfo {
 };
 
 } // namespace dxil
+
+using DXILResourceMap = MapVector<CallInst *, dxil::ResourceInfo>;
+
+class DXILResourceAnalysis : public AnalysisInfoMixin<DXILResourceAnalysis> {
+  friend AnalysisInfoMixin<DXILResourceAnalysis>;
+
+  static AnalysisKey Key;
+
+public:
+  using Result = DXILResourceMap;
+
+  /// Gather resource info for the module \c M.
+  DXILResourceMap run(Module &M, ModuleAnalysisManager &AM);
```

Comment:
DXILResourceMap is result of run.
To use DXILResourceMap more than once in different passes, do I need to call DXILResourceAnalysis::run more than once?

Reply:
No, you query the analysis manager which manages caching and invalidating the results. Something like:

```
AM.getResult<DXILResourceAnalysis>()
```